front/lib/api/files/upsert.ts

```diff
@@ -528,3 +552,97 @@ export async function processAndUpsertToDataSource(
 
   return new Ok(file);
 }
+
+function mapInternalErrorToPublicError(error: DustError): Err<
```

Comment:
This is not super desirable to have that massive map of error codes that are unrelated with the code here.

Let's instead try to unpack a DustError from the Error we get and return it directly otherwise return a generic DustError with internal_error code passing the message like it was passed before?